featuretools/entityset/entityset.py

```diff
@@ -280,231 +294,240 @@ def add_relationship(self,
-            raise ValueError(msg.format(child_v, child_e.id))
-        parent_e = relationship.parent_dataframe
-        parent_v = relationship.parent_column.id
-        if not isinstance(child_e[child_v], vtypes.Id):
-            child_e.convert_variable_type(variable_id=child_v,
-                                          new_type=vtypes.Id,
-                                          convert_data=False)
-
-        if not isinstance(parent_e[parent_v], vtypes.Index):
-            parent_e.convert_variable_type(variable_id=parent_v,
-                                           new_type=vtypes.Index,
-                                           convert_data=False)
+            raise ValueError(msg.format(child_column, child_df.ww.name))
+        parent_df = relationship.parent_dataframe
+        parent_column = relationship._parent_column_id
+        if 'foreign_key' not in child_df.ww.semantic_tags[child_column]:
+            child_df.ww.add_semantic_tags({child_column: 'foreign_key'})
+
+        if parent_df.ww.index != parent_column:
+            parent_df.ww.set_index(parent_column)
         # Empty dataframes (as a result of accessing Entity.metadata)
         # default to object dtypes for discrete variables, but
         # indexes/ids default to ints. In this case, we convert
         # the empty column's type to int
-        if isinstance(child_e.df, pd.DataFrame) and \
-                (child_e.df.empty and child_e.df[child_v].dtype == object and
-                 is_numeric_dtype(parent_e.df[parent_v])):
-            child_e.df[child_v] = pd.Series(name=child_v, dtype=np.int64)
-
-        parent_dtype = parent_e.df[parent_v].dtype
-        child_dtype = child_e.df[child_v].dtype
-        msg = u"Unable to add relationship because {} in {} is Pandas dtype {}"\
-            u" and {} in {} is Pandas dtype {}."
-        if not is_dtype_equal(parent_dtype, child_dtype):
-            raise ValueError(msg.format(parent_v, parent_e.id, parent_dtype,
-                                        child_v, child_e.id, child_dtype))
+        # --> Implementation: is this still relevant?
+        if isinstance(child_df, pd.DataFrame) and \
+                (child_df.empty and child_df[child_column].dtype == object and
+                 parent_df.ww.columns[parent_column].is_numeric):
+            child_df.ww[child_column] = pd.Series(name=child_column, dtype=np.int64)
+
+        parent_ltype = parent_df.ww.logical_types[parent_column]
+        child_ltype = child_df.ww.logical_types[child_column]
+        if parent_ltype != child_ltype:
+            warnings.warn(f'Logical type {child_ltype} for child column {child_column} does not match '
+                          f'parent column {parent_column} logical type {parent_ltype}. '
+                          'Changing child logical type to match parent.')
+            child_df.ww.set_types(logical_types={child_column: parent_ltype})
```

Comment:
Unsure if this change was already discussed, but it is changing from an error to a warn and try to continue anyway.   

If we do use this approach,  I think we should document this in release notes with the other noteworthy changes

Reply:
I think it was discussed briefly in the larger context of how `Categorical` logical type was getting used for columns that were variable type `Id`, and it was causing issues with this line because sometimes the categories in the dtype weren't matching. Somehow we ended here with comparing logical types, and I think part of that was deciding that it was okay to try converting logical types. 

Happy to document in the release notes if you're okay with leaving the behavior here.